### DISCUSSION TITLE
"App not installed?"

### DISCUSSION THREAD

**Discussion by tematikkp** [ppy/osu#21924] [Q&A] [answered]
Hi, I'm trying to update osu!lazer from 2022.1208.0 to 2022.1228.0 in mobile, and why app not installed? peppy pls help me

**Comment by bdach** [comment, marked as answer]
you may need to uninstall the previous version manually (https://github.com/ppy/osu/discussions/21895)

**Reply by DragonRider75** [reply]
I'm dealing with the same issue. But I have a quick question, if I were to manually uninstall the previous version to install this version, will I get all of the beatmaps that I previously downloaded?

**Reply by bdach** [reply]
i suppose uninstalling would delete your downloaded beatmaps, yes. maybe we want to get the signature fixed in that case, not sure.

**Reply by Shirou27** [reply]
You can just rename the data folder before uninstalling it, and rename it back after installing the new version

**Reply by peppy** [reply]
This has been resolved with the more recent releases with correct signing, for whatever that's worth.

**Reply by tematikkp** [reply]
thank you peppy it has been fixed in 2023.121.0, you're the best! I don't need to uninstall it and reinstall it

**Comment by DragonRider75** [comment]
I am still getting the same issue when updating from 2022.1228.0 to 2023.121.0, in which I keep getting an "App not installed." message. I don't know what I'm doing wrong, and I certainly do not want to uninstall and then reinstall. Please help. 
![Screenshot_20230122-080214_Package installer](https://user-images.githubusercontent.com/114231772/213917270-d5622896-435e-48f5-ac3f-64ecc0bc2d89.jpg)

**Reply by Shirou27** [reply]
but uninstalling is the only way if you can't install the new update. If you don't want to lose any of your music, replays, etc. Download ZArchiver and go to Android then data and find "sh.ppy.osulazer" folder, rename that folder and you can now unistall and install the new update and rename the folder back to "sh.ppy.osulazer" after installing the new update.

**Reply by DragonRider75** [reply]
I did what you said and it actually worked. Thank you.

**Reply by tematikkp** [reply]
welp it's because 2022.1228.0 has different certificates than 2023.121.0+. (peppy already said that, see also #21895)

**Reply by peppy** [reply]
We fixed that along the way, but yes, for a short period we were signing using a default certificate. The current (and old) one should continue to be used going forward, so hopefully this only affected the handful of users which downloaded the bad build before we replaced it.
